### PR TITLE
ENG-659 update docker in dind image

### PR DESCRIPTION
**Pull request by dfuentes77**
This is to support pulling down newer built containers that require signature key

### CODE REVIEW

File: dind/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-  DOCKER_COMPOSE_VERSION=1.20.1 \
+ENV DOCKER_VERSION=18.09.9 \
+  DOCKER_COMPOSE_VERSION=1.22.0 \
   DOCKER_PACKAGES="curl device-mapper mkinitfs bash e2fsprogs e2fsprogs-extra iptables python-dev" \
   OTHER="git"
 
 # Install Docker, Docker Compose
 RUN apk --update --no-cache add $DOCKER_PACKAGES $OTHER && \
-    curl https://get.docker.com/builds/Linux/x86_64/docker-${DOCKER_VERSION}.tgz | tar zx && \
+    curl https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_VERSION}.tgz | tar zx && \
```
Comment (dfuentes77) [inline dind/Dockerfile:10]: The docker download server has changed over time

File: dind/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
     mv /docker/* /bin/ && chmod +x /bin/docker* \
 && \
     apk add py-pip && \
-    pip install docker-compose==${DOCKER_COMPOSE_VERSION}
+    pip install docker-compose==${DOCKER_COMPOSE_VERSION} \
+&& \
+    docker -v && \
+    docker-compose -v
```
Comment (dfuentes77) [inline dind/Dockerfile:17]: want to display the version of docker and docker-compose as a record in the build

File: dind/Makefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 build:
-	docker build -t ceregousa/dind:docker-1.13 .
+	docker build -t ceregousa/dind:docker-18.09.9 .
```
Comment (dfuentes77) [inline dind/Makefile:2]: update the tag/name used when pushed to docker hub

File: dind/docker-lib.sh [hunk 4]
```diff
@@ -38,6 +38,11 @@ sanitize_cgroups() {
             ln -s "$mountpoint" "/sys/fs/cgroup/$sys"
         fi
     done
+
+    if [ ! -e /sys/fs/cgroup/systemd ] && [ $(cat /proc/self/cgroup | grep '^1:name=openrc:' | wc -l) -eq 0 ]; then
+      mkdir /sys/fs/cgroup/systemd
+      mount -t cgroup -o none,name=systemd none /sys/fs/cgroup/systemd
+    fi
```
Comment (dfuentes77) [inline dind/docker-lib.sh:45]: I needed this from latest version of script, in order to mount systemd path.  This allows dockerd to startup correctly with version 18.x

File: dind/docker-lib.sh [hunk 5]
```diff
@@ -65,7 +70,7 @@ start_docker() {
         server_args="${server_args} -g=$3"
     fi
 
-    docker daemon ${server_args} >/tmp/docker.log 2>&1 &
+    dockerd ${server_args} >/tmp/docker.log 2>&1 &
```
Comment (dfuentes77) [inline dind/docker-lib.sh:73]: need to call dockerd binary directly

File: dind/Dockerfile [hunk 6]
```diff
@@ -1,17 +1,20 @@
 FROM alpine:3.5
 
-ENV DOCKER_VERSION=1.13.1 \
-  DOCKER_COMPOSE_VERSION=1.20.1 \
+ENV DOCKER_VERSION=18.09.9 \
```
Comment (phrazzld) [inline dind/Dockerfile:3]: 17 major version bumps??

**Review by phrazzld — APPROVED** [review]
As long as that crazy looking version change is correct LGTM

File: dind/Dockerfile (hunk 6 above)
Comment (dfuentes77) [reply dind/Dockerfile:3]: you how it is... they just jumped versions from 1.13 to 17.x for year based versions.  So it's really only 2 major versions newer.  Still it's 2018 which is still old and I'm thinking I might try out a newer version to see if it just works

File: dind/Dockerfile [hunk 7]
```diff
@@ -1,17 +1,21 @@
-FROM alpine:3.5
+FROM alpine:3.19
```
Comment (dfuentes77) [inline dind/Dockerfile:1]: was able to use the latest version of alpine

File: dind/Dockerfile [hunk 8]
```diff
@@ -1,17 +1,21 @@
...
 
-ENV DOCKER_VERSION=1.13.1 \
-  DOCKER_COMPOSE_VERSION=1.20.1 \
-  DOCKER_PACKAGES="curl device-mapper mkinitfs bash e2fsprogs e2fsprogs-extra iptables python-dev" \
+ENV DOCKER_VERSION=19.03.9 \
+  DOCKER_COMPOSE_VERSION=1.24.1 \
+  DOCKER_PACKAGES="curl device-mapper mkinitfs bash e2fsprogs e2fsprogs-extra iptables python3-dev" \
```
Comment (dfuentes77) [inline dind/Dockerfile:5]: was able to use 3 major versions newer docker runtime and 4 minor versions newer docker-compose (which supports easier logs output)

**Pull request merged**